pyiron_atomistics/atomistics/structure/atoms.py

```diff
@@ -281,12 +276,16 @@ def set_species(self, value):
             value (list): A list atomistics.structure.periodic_table.ChemicalElement instances
 
         """
-        if value is None:
-            return
-        value = list(value)
-        self._species_to_index_dict = {el: i for i, el in enumerate(value)}
-        self._species = value[:]
-        self._store_elements = {el.Abbreviation: el for el in value}
+        if value is not None:
+            self._species = list(value)[:]
+
+    @property
+    def _store_elements(self) -> dict:
+        return {el.Abbreviation: el for el in self.species}
+
+    @property
+    def _species_to_index_dict(self) -> dict:
```

Comment:
This might trigger #718 as well, right? I guess if it's currently broken anyway, it's ok that this PR is as well, but it might anyway be an appropriate time to think about a solution.

Reply:
That’s probably true, but if I understand it correctly, it’s not something that exists in ASE, right? In that case we can think about getting rid of it in the next PR, rather than fixing it in the current implementation.